tests/apm/mod.rs

```diff
@@ -107,44 +131,45 @@ fn logging() {
 
     // Create collection started
     read_first_non_monitor_line(&mut file, &mut line);
-    assert_eq!(
+    assert_eq_ignoring_ip!(
         "COMMAND.create_collection 127.0.0.1:27017 STARTED: { create: \"logging\" }\n",
-        &line
+        line
     );
 
     // Create collection completed
     line.clear();
     read_first_non_monitor_line(&mut file, &mut line);
-    assert!(line.starts_with(
+    assert_start_ignoring_ip!(
         "COMMAND.create_collection 127.0.0.1:27017 COMPLETED: { ok: 1 } (",
-    ));
+        line
+    );
     assert!(line.ends_with(" ns)\n"));
 
     // Drop collection started
     line.clear();
     read_first_non_monitor_line(&mut file, &mut line);
-    assert_eq!(
+    assert_eq_ignoring_ip!(
         "COMMAND.drop_collection 127.0.0.1:27017 STARTED: { drop: \"logging\" }\n",
-        &line
+        line
     );
```

Comment:
An alternative implementation that might be simpler would be to introduce a `coerce_ipv4` function.

```rust
fn coerce_ipv4(val: &str) -> String {
    val.replace("[::1"], "127.0.0.1")
}
```

And then you can use the built-in assert macros:

assert_eq!(
    "COMMAND.drop_collection 127.0.0.1:27017 STARTED: { drop: \"logging\" }\n",
    coerce_ipv4(&line)   
);

Reply:
Nice! I like it!